core/cli/src/informant.rs

```diff
@@ -89,8 +89,7 @@ where C: Components {
 		if let Some((ref last_num, ref last_hash)) = last {
 			if n.header.parent_hash() != last_hash {
 				let tree_route = ::client::blockchain::tree_route(
-					#[allow(deprecated)]
-					client.backend().blockchain(),
+					|id| client.header(&id)?.ok_or(::client::error::Error::UnknownBlock(format!("Unknown block {:?}", id))),
```

Comment:
You should use `ok_or_else` to avoid allocating/formatting the string even if everything is alright.